analyses/pypi/pypi.go

```diff
@@ -0,0 +1,107 @@
+package pypi
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/rs/zerolog/log"
+)
+
+// GeneratePylist creates pylist.json from requirements.txt
+func GeneratePylist(shellPath string, manifestFilePath string) string {
+	log.Debug().Msgf("Executing: Generate Pylist")
+	crdaTempPath := "/tmp/crda/"
+	err := copyToTemp("analyses/pypi/generate_pylist.py", crdaTempPath)
+	if err != nil {
+		log.Fatal().Err(err).Msgf(err.Error())
+	}
+	err = setUpEnv(shellPath, crdaTempPath, manifestFilePath)
+	if err != nil {
+		log.Fatal().Err(err).Msgf(err.Error())
+	}
+	pathToPylist := buildDepsTree(shellPath, crdaTempPath, manifestFilePath)
+	log.Debug().Msgf("Success: Generate Pylist")
+	return pathToPylist
+}
+
+//copyToTemp moves necessary files for generating pylist.json to tmp folder
+func copyToTemp(src string, crdaTempPath string) error {
+	log.Debug().Msgf("Executing: copyToTemp")
+	absSource, _ := filepath.Abs(src)
+	sourceFileStat, err := os.Stat(absSource)
+	if err != nil {
+		return err
+	}
+	if !sourceFileStat.Mode().IsRegular() {
+		return fmt.Errorf("%s is not a regular file", src)
+	}
+	dst := crdaTempPath + sourceFileStat.Name()
+	source, err := os.Open(absSource)
+	if err != nil {
+		return err
+	}
+	defer source.Close()
+	os.MkdirAll(crdaTempPath, os.ModePerm)
+	destination, err := os.Create(dst)
+	if err != nil {
+		return err
+	}
+	defer destination.Close()
+	_, err = io.Copy(destination, source)
+	if err != nil {
+		return err
+	}
+	log.Debug().Msgf("Success: copyToTemp")
+	return err
+}
+
+// setUpEnv sets up virtual env and install dependencies.
+func setUpEnv(shellPath string, crdaTempPath string, manifestFilePath string) error {
+	log.Debug().Msgf("Executing: setUpEnv")
+	pyExecPath, err := exec.LookPath("python")
+	if err != nil {
+		log.Fatal().Err(err).Msgf("Please make sure Python v3.6+ is installed. Hint: Check same by executing `python --version`\n.")
+	}
+	os.MkdirAll(crdaTempPath, os.ModePerm)
+
+	cmdCreateVenv := &exec.Cmd{
+		Path: pyExecPath,
+		Args: []string{pyExecPath, "-m", "venv", "venv"},
+	}
+	cmdInstallDeps := &exec.Cmd{
+		Path: pyExecPath,
+		Args: []string{pyExecPath, "venv/bin/pip", "install", "--user", "-r", manifestFilePath},
+	}
+	finalCmd := fmt.Sprintf("%s && %s",
+		cmdCreateVenv.String(),
+		cmdInstallDeps.String())
+	log.Debug().Msgf("Executing command for Env. Setup %s", finalCmd)
+	cmd := exec.Command(shellPath, "-c", finalCmd)
+	cmd.Stderr = os.Stdout
+	cmd.Dir = crdaTempPath
+	if err := cmd.Run(); err != nil {
+		log.Fatal().Err(err).Msgf(err.Error())
+	}
+	log.Debug().Msgf("Success: setUpEnv")
```

Comment:
It is not recommended to install the deps. We should follow snyk here by taking a full path to python which it has virtulenv installed by the user. For more details, `synk --help`

Reply:
In CI, How likely you see this working..? 
AFAIK CI doesn't retain state, How likely we are able to find `venv` in current folder, 

PS: CI would be triggered by Webhooks, and it's highly unlikely that the user would Push `venv` to Version Control System.

Reply:
I didn't say that the venv will be part of the source code, I'm saying the venv preparation & installation of dependencies must be responsibility of the user. Thats how snyk cli work. I would recommend to give a snyk cli a try on all 4 ecosystems.

Reply:
So, CI doesn't retain state and we are expecting from Admin, each time before triggering this built, to run a separate Job of installing dependencies?

Reply:
@deepak1725 , usually all projects which runs tests under CI should install its dependencies.

Reply:
Okay np.

For that matter, where should we expect dependencies to be installed..by a separate Job?

- local (same folder as of binary) -- This can be achieved via virtual env or local installation. This would be compatible with VS Code Ext. current implementation.

- global (site_packages - system python)

Can we expect to have a flag for `site_package` Path to cmd?

Reply:
Ack. As per discussions, Virtual Env will be there in the same folder as of this binary.